src/openloco/windows/construction/OverheadTab.cpp

```diff
@@ -0,0 +1,387 @@
+
+                if (_lastSelectedMods & 1 << 1)
+                    activatedWidgets |= 1 << widx::checkbox_2;
+            }
+        }
+        else
+        {
+            auto trackObj = objectmgr::get<track_object>(_trackType);
+
+            auto args = FormatArguments();
+            args.push(trackObj->name);
+
+            if (_modList[0] != 0xFF)
+            {
+                auto trackExtraObject = objectmgr::get<track_extra_object>(_modList[0]);
+                self->widgets[widx::checkbox_1].type = widget_type::checkbox;
+                self->widgets[widx::checkbox_1].text = trackExtraObject->name;
+
+                if (_lastSelectedMods & 1 << 0)
+                    activatedWidgets |= 1 << widx::checkbox_1;
+            }
+
+            if (_modList[1] != 0xFF)
+            {
+                auto trackExtraObject = objectmgr::get<track_extra_object>(_modList[1]);
+                self->widgets[widx::checkbox_2].type = widget_type::checkbox;
+                self->widgets[widx::checkbox_2].text = trackExtraObject->name;
+
+                if (_lastSelectedMods & 1 << 1)
+                    activatedWidgets |= 1 << widx::checkbox_2;
+            }
+
+            if (_modList[2] != 0xFF)
+            {
+                auto trackExtraObject = objectmgr::get<track_extra_object>(_modList[2]);
+                self->widgets[widx::checkbox_3].type = widget_type::checkbox;
+                self->widgets[widx::checkbox_3].text = trackExtraObject->name;
+
+                if (_lastSelectedMods & 1 << 2)
+                    activatedWidgets |= 1 << widx::checkbox_3;
+            }
+
+            if (_modList[3] != 0xFF)
+            {
+                auto trackExtraObject = objectmgr::get<track_extra_object>(_modList[3]);
+                self->widgets[widx::checkbox_4].type = widget_type::checkbox;
+                self->widgets[widx::checkbox_4].text = trackExtraObject->name;
+
+                if (_lastSelectedMods & 1 << 3)
+                    activatedWidgets |= 1 << widx::checkbox_4;
+            }
+        }
+
+        self->activated_widgets = activatedWidgets;
+
+        self->widgets[widx::image].type = widget_type::none;
+        self->widgets[widx::track].type = widget_type::none;
+        self->widgets[widx::track_dropdown].type = widget_type::none;
+
+        self->widgets[widx::image].tooltip = string_ids::null;
+
+        if (_lastSelectedMods & 0xF)
+        {
+            self->widgets[widx::image].type = widget_type::wt_3;
+            self->widgets[widx::track].type = widget_type::wt_18;
+            self->widgets[widx::track_dropdown].type = widget_type::wt_11;
+
+            self->widgets[widx::image].tooltip = string_ids::upgrade_track_with_mods;
+
+            if (is_unknown_3_mode())
+            {
+                if (_toolWindowType == WindowType::construction)
+                    self->widgets[widx::image].tooltip = string_ids::click_track_to_upgrade;
```

Comment:
I think this gives a strong indication as to what mode 3 is.

Reply:
trackUpgadeMode?

Reply:
Yeah that would be better than `unknown_3`